DecipherNow/helm-charts
# [release-2.2] Control values to map

**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.


This PR changes the global values `control_port`, `control_cluster_label` and `control_additional_namespaces` to 

```
global:
  control:
     port:
     cluster_label:
     additional_namespaces:
```

closes #586 

<br/><br/>

2. What **changes to custom.yaml** are required?

The structure needs to be updated from 

```yaml
  # Port for Grey Matter Control. Used in catalog init and sidecar envvars.
  control_port: 50000
  # The label Control uses to find pods to include in the mesh
  control_cluster_label: greymatter.io/control
```

to

```yaml
  control:
    # Port for Grey Matter Control. Used in sidecar envvars
    port: 50000
    # The label Control uses to find pods to include in the mesh
    cluster_label: greymatter.io/control
```


<br/><br/>

3. Have you documented any additional setup steps or configurations options?

None
<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

To test:

`make install`

you can turn off spire in global.yalm and `make install` to test without spire.

If you want, change the label at `global.control.cluster_label`, then `make install` and ensure that everything comes up and is discovered properly with the new label.

<br/><br/>
